File: blender_extension/roblox_interface/roblox_server/format_types/pos2d.py
import struct

format = struct.Struct("<2f")
format_size = format.size
unpack_from = format.unpack_from
pack = format.pack

def read(self, format_data, **masks):
	return unpack_from(*self.read_buffer(format_size))

def write(self, args, args_count, format_data, **masks):
	pos_value = args[args_count]
	self.write_buffer(pack(pos_value.x, pos_value.y), format_size)
	return args_count + 1

File: roblox_plugin/src/Server/formatTypes/pos2d.ts
//!native
import type * as ProcessFormats from "../processFormats";
import SendMessagesThread from "../sendMessages";
import ReceiveMessagesThread from "../receiveMessages";

const DATA_SIZE = 8;
export function read(
	receiveThread: ReceiveMessagesThread,
	args: defined[],
	formatData: ProcessFormats.FormatData,
	masks: Map<string, boolean>,
) {
	const [readBuffer, readOffset] = ReceiveMessagesThread.readBuffer(receiveThread, DATA_SIZE);
	const x = buffer.readf32(readBuffer, readOffset);
	const y = buffer.readf32(readBuffer, readOffset + 4);
	args.push(new Vector2(x, y));
}
export function write(
	sendThread: SendMessagesThread,
	args: defined[],
	argsCount: number,
	formatData: ProcessFormats.FormatData,
	masks: Map<string, boolean>,
): number {
	const value = args[argsCount] as Vector2;

	const data = buffer.create(DATA_SIZE);
	buffer.writef32(data, 0, value.X);
	buffer.writef32(data, 4, value.Y);
	SendMessagesThread.writeBuffer(sendThread, data, DATA_SIZE);
	return argsCount + 1;
}
